app/src/services/mongo/MongoDBConnection.ts

```diff
@@ -0,0 +1,79 @@
+import mongoose, {Mongoose} from 'mongoose';
+import ApplicationException from '../../exceptions/ApplicationException';
+import {getLogger} from '../../logger';
+import {injectable} from 'inversify';
+import Database, {DatabaseConfig} from '../../config/database';
+
+@injectable()
+export default class MongoDBConnection {
+    private static readonly logger = getLogger('DatabaseConnection');
+    private static isConnected = false;
```

Comment:
Static stuff is going against the principle of IoC.

Reply:
Suggestions for fix? Was really struggling with this one when it came to accessing properties within promises

Reply:
Will have a look tomorrow, but it is mostly about removing the static keywords so every new instance can have its own state.

Reply:
Proxying to issue for now, merging code as everything else is resolved